peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
 
-However, often the "cooked" string is what is needed, by decoding the string as
-if it were a standard Python string. In the proposed implementation, the decoded object's
-``__new__`` will *store* the raw string and *store and return* the "cooked" string.
+See the `Common Patterns Seen in Processing Templates`_ section for more
+information on how to process template strings. See the `Examples`_ section
+for detailed working examples.
 
-The protocol is marked as ``@runtime_checkable`` to allow structural pattern matching to
-test against the protocol instead of a type. This can incur a small performance penalty.
-Since the ``case`` tests are in user-code tag functions, authors can choose to optimize by
-testing for the implementation type discussed next.
 
-The ``Decoded`` protocol will be available from ``typing``. In CPython, ``Decoded``
-will be implemented in C, but for discussion of this PEP, the following is a compatible
-implementation:
-
-.. code-block:: python
+No Support for Concatenation
+----------------------------
 
-    class DecodedConcrete(str):
-        _raw: str
-
-        def __new__(cls, raw: str):
-            decoded = raw.encode("utf-8").decode("unicode-escape")
-            if decoded == raw:
-                decoded = raw
-            chunk = super().__new__(cls, decoded)
-            chunk._raw = raw
-            return chunk
-
-        @property
-        def raw(self):
-            return self._raw
+Because template strings are designed to be used in custom DSLs, they do not
+support explicit concatenation using ``+``. This is unlike all other string
+literals in Python today. Concatenation may have different semantics in 
+different DSLs or no meaningful semantics at all; accordingly, the ``Template`` 
+type that t-strings evaluate to does not support ``__add__`` or similar methods.
 
-Interpolation
--------------
+The following code will raise a ``TypeError``:
 
-An ``Interpolation`` is the data structure representing an expression inside the tag
-string. Interpolations enable a delayed evaluation model, where the interpolation
-expression is computed, transformed, memoized, or processed in any way.
+.. code-block:: python
 
-In addition, the original text of the interpolation expression is made available to the
-tag function. This can be useful for debugging or metaprogramming.
+    t"Hello " + t"World"
 
-``Interpolation`` is a ``Protocol`` which will be made available from ``typing``. It
-has the following definition:
+Implicit concatenation is also not supported. The following code will raise 
+a ``SyntaxError``:
 
 .. code-block:: python
 
-    @runtime_checkable
-    class Interpolation(Protocol):
-        def __len__(self):
-            ...
+    t"Hello " "World"
 
-        def __getitem__(self, index: int):
-            ...
 
-        def getvalue(self) -> Callable[[], Any]:
-            ...
+Support for the Debug Specifier
+-------------------------------
```

Comment:
It wasn't immediatley clear what the debug specifier was from the title, perhaps:
```suggestion
Support for the debug specifier (``=``)
---------------------------------------
```